### PR TITLE
Exo words question

### PR DESCRIPTION
**Pull request by ngodfraind**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | yes/no
| Fixed tickets | comma-separated list of tickets fixed by the PR, if any

### CODE REVIEW

File: plugin/exo/Resources/less/items/words.less [hunk 1]
```diff
@@ -0,0 +1,29 @@
+.word-success {
```
Comment (stefk) [inline plugin/exo/Resources/less/items/words.less]: Please remove the rules related to words in quiz/editor/style.css if don't use them.

File: plugin/exo/Resources/less/items/words.less [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  color: @brand-success
+}
+
+.word-danger {
+  color: @brand-danger
+}
+
+.choice-paper {
```
Comment (stefk) [inline plugin/exo/Resources/less/items/words.less]: You're overriding rules of another question type.

File: plugin/exo/Resources/modules/items/words/paper.jsx [hunk 3]
```diff
@@ -1,6 +1,122 @@
...
 
-export const WordsPaper = () =>
-  <div>
-    Words paper
-  </div>
+function getClassNameForSolution(solution) {
+  return solution.found ?
+    solution.score > 0 ? 'bg-success text-success' : 'bg-danger text-danger' :
+    ''
+}
```
Comment (stefk) [inline plugin/exo/Resources/modules/items/words/paper.jsx]: Not needed and would be more readable if lines 27-29 were:

```jsx
className={classes('item', {
  'bg-success text-success': el.found && el.score > 0,
  'bg-danger text-danger': el.found && el.score <= 0
})}
```

File: plugin/exo/Resources/modules/items/words/utils/utils.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  return split
+}
+
+utils.getTextElements = (text, solutions) => {
+  const matches = solutions.map(solution => solution.text)
+  const data = []
+
+  //first we find each occurence of a given word
+  matches.forEach(word => {
+    let regex = new RegExp('\\b' + word + '\\b', 'gi')
```
Comment (stefk) [inline plugin/exo/Resources/modules/items/words/utils/utils.js]: Should be const

File: plugin/exo/Resources/modules/items/words/utils/utils.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    let position = text.search(regex)
```
Comment (stefk) [inline plugin/exo/Resources/modules/items/words/utils/utils.js]: Same

File: plugin/exo/Resources/modules/items/words/utils/utils.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      word,
+      position,
+      score: solutions.find(el => el.text === word).score,
+      feedback: solutions.find(el => el.text === word).feedback,
+      found: position > -1
+    })
+  })
+
+  return data
+}
```
Comment (stefk) [inline plugin/exo/Resources/modules/items/words/utils/utils.js]: `utils.split` really needs a test, even a simple one.

File: plugin/exo/Resources/modules/items/words/utils/utils.js (hunk 4 above)
Comment (Elorfin) [inline plugin/exo/Resources/modules/items/words/utils/utils.js]: The `i` modifier should be applied according to the keyword `caseSensitive` property.

File: plugin/exo/Resources/modules/items/words/utils/utils.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  return split
+}
+
+utils.getTextElements = (text, solutions) => {
+  const matches = solutions.map(solution => solution.text)
+  const data = []
+
+  //first we find each occurence of a given word
+  matches.forEach(word => {
+    const regex = new RegExp('\\b' + word + '\\b', 'gi')
```
Comment (Elorfin) [inline plugin/exo/Resources/modules/items/words/utils/utils.js]: The `i` modifier should be applied according to the keyword `caseSensitive` property.

**Pull request merged**